### PR TITLE
chore(webpack-cli): reduce install size

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
chore

**Did you add tests for your changes?**

**If relevant, did you update the documentation?**

**Summary**
remove unused deps and reduce install size.

**Does this PR introduce a breaking change?**
No

**Other information**
No

### CODE REVIEW

File: packages/webpack-cli/__tests__/info.test.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//eslint-disable-next-line node/no-extraneous-require
```
Comment (evilebottnawi) [inline packages/webpack-cli/__tests__/info.test.js:1]: I think we need to improve config for this rule

Comment (snitin315) [reply packages/webpack-cli/__tests__/info.test.js:1]: Hmm, we have disabled this rule at many places.

Comment (evilebottnawi) [reply packages/webpack-cli/__tests__/info.test.js:1]: Yep, it is not good, let's improve this

Comment (snitin315) [reply packages/webpack-cli/__tests__/info.test.js:1]: I think we can `disable` it for test files.

Comment (evilebottnawi) [reply packages/webpack-cli/__tests__/info.test.js:1]: @snitin315 Ideally we have all tests packages in devDeps, so it should be not triggered, need investigate

Comment (snitin315) [reply packages/webpack-cli/__tests__/info.test.js:1]: Maybe we need to use [`allowModules`](https://github.com/mysticatea/eslint-plugin-node/blob/master/docs/rules/no-extraneous-require.md#allowmodules) ->

```js
'node/no-extraneous-require': ['error', {
            'allowModules': ['execa', 'webpack', 'webpack-dev-server']
        }]
```

Comment (evilebottnawi) [reply packages/webpack-cli/__tests__/info.test.js:1]: Why do warning this rule? We have this in our dev deps

Comment (snitin315) [reply packages/webpack-cli/__tests__/info.test.js:1]: No idea why it's triggering 😢

Comment (evilebottnawi) [reply packages/webpack-cli/__tests__/info.test.js:1]: Can you look at rule logic?

Comment (snitin315) [reply packages/webpack-cli/__tests__/info.test.js:1]: This one is not specified explicitly in rules.
https://github.com/webpack/webpack-cli/blob/2c390a1fdcbb85b7596c7b621c8d263da1734690/.eslintrc.js#L11-L18

Comment (evilebottnawi) [reply packages/webpack-cli/__tests__/info.test.js:1]: I mean inside code of rule and check why we got error here

Comment (snitin315) [reply packages/webpack-cli/__tests__/info.test.js:1]: https://github.com/mysticatea/eslint-plugin-node/blob/4a348e7d24e0d806d7873f293c91a922d0316d14/lib/util/check-extraneous.js#L27-L34

I logged the dependencies set here and found its collecting deps defined inside `packages/webpack-cli/package.json` and not root `package.json` -

Screenshot at 2020-10-09 15-53-40

Comment (snitin315) [reply packages/webpack-cli/__tests__/info.test.js:1]: I think we should create an issue for the same in the `eslint-plugin-node` repo.

Comment (evilebottnawi) [reply packages/webpack-cli/__tests__/info.test.js:1]: Oh, I think we need setup `resolvePaths` (https://github.com/mysticatea/eslint-plugin-node/blob/master/docs/rules/no-extraneous-require.md, for `import` too) and setup root `node_modules` here

Comment (evilebottnawi) [reply packages/webpack-cli/__tests__/info.test.js:1]:
```
 "settings": {
        "node": {
            "resolvePaths": [__dirname],
        }
    },
```

let try in `eslint` root config

Comment (snitin315) [reply packages/webpack-cli/__tests__/info.test.js:1]: still the same :confused:

Comment (evilebottnawi) [reply packages/webpack-cli/__tests__/info.test.js:1]: Can you check different values and look at source code of the rule

Comment (snitin315) [reply packages/webpack-cli/__tests__/info.test.js:1]: In near future 👍

**Comment by snitin315** [comment]
Closed due to conflicts, will open a new one.

**Pull request closed without merging**